Packs/TrendMicroEmailSecurity/Integrations/TrendMicroEmailSecurityEventCollector/TrendMicroEmailSecurityEventCollector.py

```diff
@@ -0,0 +1,488 @@
+
+""" CONSTANTS """
+
+VENDOR = "trend_micro"
+PRODUCT = "email_security"
+DATE_FORMAT_EVENT = "%Y-%m-%dT%H:%M:%SZ"
+
+
+class EventType(str, Enum):
+    ACCEPTED_TRAFFIC = "accepted_traffic"
+    BLOCKED_TRAFFIC = "blocked_traffic"
+    POLICY_LOGS = "policy_logs"
+
+
+URL_SUFFIX = {
+    EventType.ACCEPTED_TRAFFIC: "/api/v1/log/mailtrackinglog",
+    EventType.BLOCKED_TRAFFIC: "/api/v1/log/mailtrackinglog",
+    EventType.POLICY_LOGS: "/api/v1/log/policyeventlog",
+}
+
+
+""" CLIENT CLASS """
+
+
+class NoContentException(Exception):
+    """
+    Error definition for API response with status code 204
+    Makes it possible to identify a specific exception
+    that arises from the API and to handle this case correctly
+    see `handle_error_no_content` method
+    """
+
+    ...
+
+
+class Client(BaseClient):
+    """Client class to interact with the service API
+
+    :param base_url (str): server url.
+    :param username (str): the account username.
+    :param api_key (str): the account api_key.
+    :param verify (bool): specifies whether to verify the SSL certificate or not.
+    :param proxy (bool): specifies if to use XSOAR proxy settings.
+    """
+
+    def __init__(
+        self, base_url: str, username: str, api_key: str, verify: bool, proxy: bool
+    ):
+        authorization_encoded = self._encode_authorization(username, api_key)
+        headers = {"Authorization": f"Basic {authorization_encoded}"}
+
+        super().__init__(base_url=base_url, verify=verify, proxy=proxy, headers=headers)
+
+    def _encode_authorization(self, username: str, api_key: str) -> str:
+        authorization_bytes = f"{username}:{api_key}".encode()
+        return base64.b64encode(authorization_bytes).decode()
+
+    def get_logs(self, event_type: EventType, params: dict):
+        return self._http_request(
+            "GET",
+            url_suffix=URL_SUFFIX[event_type],
+            params=params,
+            error_handler=self.handle_error_no_content,
+            ok_codes=(200, 201, 202, 203),
+        )
+
+    def handle_error_no_content(self, res) -> None:
+        if res.status_code == 204:
+            raise NoContentException("No content")
+        self.client_error_handler(res)
+
+
+""" HELPER FUNCTIONS """
+
+
+def add_missing_fields_to_event(event: dict):
+    for field in (
+        "action",
+        "mailID",
+        "sender",
+        "genTime",
+        "logType",
+        "subject",
+        "tlsInfo",
+        "senderIP",
+        "direction",
+        "eventType",
+        "messageID",
+        "recipient",
+        "domainName",
+        "headerFrom",
+        "policyName",
+        "eventSubtype",
+        "policyAction",
+        "deliveredTo",
+        "attachments",
+        "recipients",
+        "headerTo",
+        "details",
+        "timestamp",
+        "size",
+        "deliveryTime",
+        "reason",
+        "embeddedUrls",
+    ):
+        if field not in event:
+            event[field] = None
```

Comment:
can replace with event |= {key: None for key in set(<put all keys here>).difference(event.keys())}

Reply:
@dorschw @israelpoli 
actually I am concerned about what is the fastest way to do this.

Reply:
```python
from random import sample
import timeit
from collections import Counter

from tqdm import tqdm

all_keys = tuple(range(1000))

def update():
    partial_dict = {i:None for i in sample(all_keys,500)}
    partial_dict |= {k: None for k in set(all_keys).difference(partial_dict.keys())}

def for_loop():
    partial_dict = {i:None for i in sample(all_keys,500)}
    for k in all_keys:
        if k not in partial_dict:
            partial_dict[k] = None
    
def for_loop_set():
    partial_dict = {i:None for i in sample(all_keys,500)}
    for k in set(all_keys).difference(partial_dict.keys()):
        partial_dict[k] = None

times = 100

results = []

for _ in tqdm(range(300)):
    for_loop_time = timeit.timeit(for_loop, number=times)
    for_loop_set_time = timeit.timeit(for_loop, number=times)
    update_time = timeit.timeit(update, number=times)

    # print(f"{for_loop_time=}")
    # print(f"{update_time=}")
    # print(f"{for_loop_set_time=}")
    
    best = min((for_loop_time,for_loop_set_time,update_time))
    if for_loop_time == best:
        result = "for"
    elif for_loop_set_time == best:
        result = "for set"
    elif update_time == best:
        result = "update"
    results.append(result) 

print(Counter(results))
```
```python
Counter({'for set': 121, 'for': 96, 'update': 83})
```

I'll go with a for-on-set